static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx

```diff
@@ -597,9 +611,9 @@ export class TraceTree {
       }
 
       if (
-        childrenCount === 1 &&
         isPageloadTransactionNode(node) &&
-        isServerRequestHandlerTransactionNode(parent)
+        isServerRequestHandlerTransactionNode(parent) &&
+        getPageloadTransactionChildCount(parent) === 1
```

Comment:
just for me to understand: We don't do anything if the parent (http.server) txn has more than one `op: pageload` child txns, right?

Reply:
Exactly (afaik this should never happen), but the main reason being that we dont do anything is because we dont know which pageload to reparent the server handler under.